source/common/http/http1/codec_impl.cc

```diff
@@ -505,16 +544,18 @@ void ConnectionImpl::onHeaderValue(const char* data, size_t length) {
   const uint32_t total =
       current_header_field_.size() + current_header_value_.size() + current_header_map_->byteSize();
   if (total > (max_headers_kb_ * 1024)) {
-
+    const std::string header_type = processing_trailers_ ? "trailers" : "headers";
     error_code_ = Http::Code::RequestHeaderFieldsTooLarge;
     sendProtocolError(Http1ResponseCodeDetails::get().HeadersTooLarge);
-    throw CodecProtocolException("headers size exceeds limit");
+    throw CodecProtocolException(absl::StrCat(header_type, " size exceeds limit"));
   }
 }
 
 int ConnectionImpl::onHeadersCompleteBase() {
-  ENVOY_CONN_LOG(trace, "headers complete", connection_);
+  const std::string header_type = processing_trailers_ ? "trailers" : "headers";
```

Comment:
Make this a string_view here and in similar cases, and define the string_view constants somewhere global.